flocker/node/_deploy.py

```diff
@@ -594,6 +609,7 @@ def discover_state(self, local_state):
             # convergence agent for datasets will discover the information
             # and then we can proceed.
             return succeed([NodeState(
+                uuid=self.uuid,
```

Comment:
Looks like it probably is.  `flocker.node.test.test_deploy.ApplicationNodeDeployerDiscoverNodeConfigurationTests.test_unknown_manifestations` asserts a `NodeState` with the right uuid comes back.